api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -1160,6 +1160,10 @@ public static ListType ofRequired(int elementId, Type elementType) {
     private transient List<NestedField> fields = null;
 
     private ListType(NestedField elementField) {
+      Preconditions.checkArgument(
+          !elementField.type().equals(UnknownType.get()),
+          "Cannot create ListType with unknown type: %s",
+          elementField.name);
```

Comment:
I'm not sure that we should fail here. Can't Avro represent a list of unknowns using a `NULL` schema?

I would rather fail in the Parquet integration when converting to a Parquet schema because this is not allowed in Parquet but is fine in Avro or Iceberg.

Reply:
> Can't Avro represent a list of unknowns using a `NULL` schema?

Yes, we can do that, but then we still store the `NullType` in the Avro schema. And I think that's not in line what's in the spec (unless I have a different interpretation of it):

<img width="710" height="371" alt="image" src="https://github.com/user-attachments/assets/919142a7-71bf-49b7-989f-af198a5c1985" />

When we also omit the Avro `Array`, then we have the nullability problem as mentioned on the devlist.